CRM/Core/BAO/CustomGroup.php

```diff
@@ -1832,11 +1832,15 @@ public static function formatGroupTree(&$groupTree, $groupCount = 1, &$form = NU
     $formattedGroupTree = array();
     $uploadNames = $formValues = array();
 
-    // JSON encoded submitted form values passed via URL
-    $submittedValues = CRM_Utils_Request::retrieve('post', 'String');
-    if (!empty($submittedValues)) {
-      $submittedValues = json_decode($submittedValues, TRUE);
-    }
+    // Generate cache key against which submitted custom value will be stored
+    $cacheKey = implode('_',
+      array(__CLASS__,
```

Comment:
Ya I tried with qfKey earlier but getting some trouble with it as formatGroupTree() function is being called by 'CRM_Custom_Form_CustomDataByType' form which is called via loadPage and entity form. So qfkey keeps changing for this two form instance. I tried to use ```get_class($form) != 'CRM_Custom_Form_CustomDataByType'``` as another flag to set/get qfkey but it throws invalid session handling error on page reload. Still lemme check if we can make use of gfkey.